pkg/kedge/grpc/director/router/router.go

```diff
@@ -88,15 +98,33 @@ func (r *static) serviceNameMatches(fullMethodName string, matcher string) bool
 	return fullMethodName == matcher
 }
 
-func (r *static) authorityMatches(md metautils.NiceMD, matcher string) bool {
-	if matcher == "" {
+func (r *static) authorityHostMatches(md metautils.NiceMD, hostMatcher string) bool {
+	if hostMatcher == "" {
+		return true
+	}
+	auth := md.Get(":authority")
+	if auth == "" {
+		return false // there was no authority header and it was expected
+	}
+
+	return stripPort(auth) == hostMatcher
```

Comment:
Can you use net.SplitHostPort and also in authorityPortMatches?

Reply:
please no (: I tried that, but it expects port to be there (and returns error)

Reply:
What about 
```
u, e := url.Parse("null://" + auth)
if e := nil {logger.WithError(...); return false}
return u.Hostname() == hostMatcher // and similar for port
```

Reply:
I would rather avoid these hacks. I added test case for IPv6. My functions seems to work just fine (: Is that enough?